### PR TITLE
Feedback button wrapper with API

### PR DESCRIPTION
**Pull request by Troff8**
## PR includes

- [x] Feature

## Related issues

Resolve [#719 ](https://github.com/taskany-inc/issues/issues/719)

## QA Instructions, Screenshots, Recordings
  
image
image

### CODE REVIEW

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        formState: { errors, isSubmitted },
+    } = useForm<CreateFeedback>({
+        resolver: zodResolver(createFeedbackSchema),
+    });
+
+    const errorsResolver = errorsProvider(errors, isSubmitted);
+
+    const onPending = useCallback((form: CreateFeedback) => {
+        setFormBusy(true);
+        fetch('/api/feedback', {
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:27]: should be a constant

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            method: 'POST',
+            body: JSON.stringify({
+                title: form.title,
+                description: form.description,
+                href: window.location.href,
+            }),
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:33]: this fetch can fail and you should inform the user about that

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        });
+        dispatchModalEvent(ModalEvent.FeedbackCreateModal)();
+    }, []);
+
+    const onError = useCallback((err: typeof errors) => {
+        // TODO: Sentry event
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:39]: why a todo, it's a single line

File: src/pages/api/feedback.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+export const config = {
+    api: {
+        bodyParser: false,
+    },
+};
+
+const cors = Cors({
+    origin: false,
+    methods: ['HEAD', 'OPTIONS', 'POST'],
+});
```
Comment (9teen90nine) [inline src/pages/api/feedback.ts:16]: you don't need this

File: src/pages/api/feedback.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+const route = nextConnect({
+    onError(error: Error, _, res: NextApiResponse) {
+        res.status(500).json({ error: `Something went wrong: ${error.message}` });
+    },
+    onNoMatch: (_, res: NextApiResponse) => {
+        res.status(400).json({ error: 'We are sorry, but it is impossible' });
+    },
+}).use(cors);
```
Comment (9teen90nine) [inline src/pages/api/feedback.ts:25]: this is not needed

File: src/pages/api/feedback.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        let rawData = '';
+        req.on('data', (chunk: string) => {
+            rawData += chunk;
+        });
+        req.on('end', () => {
+            resolve(rawData);
+        });
+        req.on('error', (err: any) => {
+            reject(err);
+        });
```
Comment (9teen90nine) [inline src/pages/api/feedback.ts:38]: await req.json();

File: src/pages/api/feedback.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    parsedBody.userAgent = req.headers['user-agent'];
+
+    const session = await getSession({ req });
+    if (session) {
+        const { name, email, image } = session.user;
+        parsedBody.name = name;
+        parsedBody.email = email;
+        parsedBody.avatarUrl = image;
+    }
+    await fetch('https://webhook-gw.dev.app.sberdevices.ru/goals/mattermost/hooks/afaugkbry3gotf8bd9zs9d8hzy', {
```
Comment (9teen90nine) [inline src/pages/api/feedback.ts:51]: this should be an ENV variable

**Review by 9teen90nine — CHANGES_REQUESTED** [review]

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx (hunk 2 above)
Comment (Troff8) [reply src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:33]: how?

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx (hunk 3 above)
Comment (Troff8) [reply src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:39]: this is a complete copy of FilterCreateForm. what should be added here?

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                },
+                body: JSON.stringify({
+                    title: form.title,
+                    description: form.description,
+                    href: window.location.href,
+                }),
+            });
+            if (res.status < 400) {
+                dispatchModalEvent(ModalEvent.FeedbackCreateModal)();
+                dispatchSuccessNotification('sentFeedback');
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:44]: shouldn't it be a success text from locale?

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+            } else {
+                throw new Error('Request error', { cause: res });
+            }
+        } catch (e) {
+            dispatchErrorNotification('error');
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:49]: and this also

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        }
+        setFormBusy(false);
+    }, []);
+
+    const onError = useCallback((err: typeof errors) => {
+        dispatchErrorNotification('error');
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:55]: and this

File: src/components/NotificationsHub/NotificationsHub.i18n/en.json [hunk 11]
```diff
@@ -30,5 +30,6 @@
     "We are creating something new": "We are creating something new",
     "Voila! It's here 🎉": "Voila! It is here 🎉",
     "We are updating project settings": "We are updating project settings",
-    "So sad! Project will miss you": "So sad! Project will miss you"
+    "So sad! Project will miss you": "So sad! Project will miss you",
+    "feedback sent 🎉": "feedback sent 🎉"
```
Comment (9teen90nine) [inline src/components/NotificationsHub/NotificationsHub.i18n/en.json:34]: Feedback should be capitalized

File: src/components/NotificationsHub/NotificationsHub.i18n/ru.json [hunk 12]
```diff
@@ -30,5 +30,6 @@
     "We are creating something new": "Погоди немного, создаем кое-что новое...",
     "Voila! It's here 🎉": "Ура! Всё готово 🎉",
     "We are updating project settings": "Обновляем настройки",
-    "So sad! Project will miss you": "Как жаль! Проект будет скучать по тебе..."
+    "So sad! Project will miss you": "Как жаль! Проект будет скучать по тебе...",
+    "feedback sent 🎉": "Отзыв отправлен 🎉"
```
Comment (9teen90nine) [inline src/components/NotificationsHub/NotificationsHub.i18n/ru.json:34]: and here

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+                    placeholder={tr('Feedback title')}
+                    flat="bottom"
+                    brick="right"
+                    error={errorsResolver('title')}
+                />
+
+                <FormTextarea
+                    {...register('description')}
+                    minHeight={100}
+                    placeholder={tr('Feedback description. Say anythink what on your mind')}
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:73]: anything what's

File: src/components/FeedbackCreateForm/FeedbackCreateForm.i18n/en.json [hunk 14]
```diff
@@ -0,0 +1,6 @@
+{
+    "Feedback title": "",
+    "Feedback description. Say anythink what on your mind": "",
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.i18n/en.json:3]: anythinG what'S

File: src/components/FeedbackCreateForm/FeedbackCreateForm.i18n/ru.json [hunk 15]
```diff
@@ -0,0 +1,6 @@
+{
+    "Feedback title": "Заголовок заявки",
+    "Feedback description. Say anythink what on your mind": "Описание заявки. Расскажите что произошло",
```
Comment (9teen90nine) [inline src/components/FeedbackCreateForm/FeedbackCreateForm.i18n/ru.json:3]: and here

**Review by 9teen90nine — CHANGES_REQUESTED** [review]
some final quirks

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx (hunk 8 above)
Comment (Troff8) [reply src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:44]: elsewhere in the project code dispatchSuccessNotificatio is used without 'tr'
we have NotificationHub.map - https://github.com/taskany-inc/issues/blob/main/src/components/NotificationsHub/NotificationHub.map.ts

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx (hunk 9 above)
Comment (Troff8) [reply src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:49]: elsewhere in the project code dispatchErrorNotification is used without 'tr'
we have NotificationHub.map - https://github.com/taskany-inc/issues/blob/main/src/components/NotificationsHub/NotificationHub.map.ts

File: src/components/FeedbackCreateForm/FeedbackCreateForm.tsx (hunk 10 above)
Comment (Troff8) [reply src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:55]: elsewhere in the project code dispatchErrorNotification is used without 'tr'
we have NotificationHub.map - https://github.com/taskany-inc/issues/blob/main/src/components/NotificationsHub/NotificationHub.map.ts

Comment (9teen90nine) [reply src/components/FeedbackCreateForm/FeedbackCreateForm.tsx:55]: then you should use a handler like that - `const [res] = await notifyPromise(createMutation.mutateAsync(form), 'goalsCreate');`

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**